src/main/java/authentication/User.java

```diff
@@ -0,0 +1,25 @@
+package authentication;
+
+import org.example.models.tables.pojos.Professor;
+import org.example.models.tables.pojos.Student;
+
+public class User {
+    private Professor professor;
+    private Student student;
+
+    public User(Professor professor) {
+        this.professor = professor;
```

Comment:
Maybe instead of just storing the object you could extract the common properties (e.g. password)?